packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -172,10 +173,76 @@ fn.addPermission('my-service Invocation', {
 });
 ```
 
-For more information, see [Resource-based
-policies](https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html)
+You can also restrict permissions given to AWS services by providing
+a source account or ARN (representing the account and identifier of the resource
+that accesses the function or layer).
+
+For more information, see
+[Granting function access to AWS services](https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html#permissions-resource-serviceinvoke)
 in the AWS Lambda Developer Guide.
 
+### Grant function access to an AWS Organization
+
+```ts
+// Grant permissions to an entire AWS organization
+declare const fn: lambda.Function;
+const orgPrincipal = new iam.OrganizationPrincipal('o-xxxxxxxxxx');
+
+fn.grantInvoke(orgPrincipal);
+
+// Equivalent to:
+fn.addPermission('grant to entire org', {
+  principal: orgPrincipal,
+});
+
+// Equivalent to:
+fn.addPermission('grant to entire org', {
+  principal: new iam.StarPrincipal(),
+  principalOrg: orgPrincipal,
+})
+```
+
+In the above example, the `principal` will be `*` and all users in the
+organization `o-xxxxxxxxxx` will get function invocation permissions.
+
+You can restrict permissions given to the organization by specifying an
+AWS account or role as the `principal`:
+
+```ts
+// Grant permission to an account ONLY IF they are part of the organization
+declare const fn: lambda.Function;
+
+fn.addPermission('grant to account in org', {
+  principal: new iam.AccountPrincipal('123456789012'),
+  principalOrg: new iam.OrganizationPrincipal('o-xxxxxxxxxx'),
+});
+```
+
+For more information, see
+[Granting function access to an organization](https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html#permissions-resource-xorginvoke)
+in the AWS Lambda Developer Guide.
+
+### Grant function access to other AWS accounts
+
+```ts
+// Grant permission to other AWS account
+declare const fn: lambda.Function;
+const account = new iam.AccountPrincipal('123456789012');
+
+fn.grantInvoke(account);
+
+// Equivalent to:
+fn.addPermission('grant to account in org', {
```

Comment:
Let's not show the `addPermission` line. We want to train people to use the grants.